### ISSUE TITLE
Gatsby-plugin-sharp memory leaks?

### ISSUE THREAD

**Issue by chiedo** [gatsbyjs/gatsby#1511]
I've moved all of my sites to Gatsby 1.0 this week and with my largest site chiedolabs.com, I keep running into what I think are memory leaks. I have a hundred blog posts or so with large images that Gatsby is processing and if I try to run too many of those images through sharp I get something like this

```
(sharp:66579): GLib-CRITICAL **: g_hash_table_lookup: assertion 'hash_table != NULL' failed

(sharp:66579): GLib-CRITICAL **: g_hash_table_lookup: assertion 'hash_table != NULL' failed

(sharp:66579): GLib-CRITICAL **: g_hash_table_lookup: assertion 'hash_table != NULL' failed
```

Not sure where to even start debugging but for now I've just been creating versions of the plugins that bypass sharp entirely and just copy the images from point a to point b.

**Comment by chiedo** [comment]
Tracing this back to the Sharp library. Could be a result of this?

https://github.com/lovell/sharp/issues/182

I may have wrongly diagnosed. Maybe with my large site there is one file incorrectly formatted? If this is the cause, maybe we just need a try catch statement?

Will follow up as more progress is made.

**Comment by KyleAMathews** [comment]
Hmm yeah, that could be it. I'm not entirely sure it's possible to catch this but it'd be nice to give a better error for sure. If possible with the name of the file.

**Comment by chiedo** [comment]
Awesome @KyleAMathews 

I'll do some digging, try to reproduce and wrap something in a try-catch statement.

**Comment by KyleAMathews** [comment]
This will be fixed once Sharp 0.19 is out

**Comment by jongold** [comment]
Any ideas on how to work around this issue? It's intermittently breaking my builds and the only thing I can do is to continuously rebuild until it works…

Relevant dependencies:
```
    "gatsby": "^1.9.146",
    "gatsby-plugin-sharp": "^1.6.24",
    "gatsby-remark-images": "^1.5.36",
    "gatsby-transformer-remark": "^1.7.26",
    "gatsby-transformer-sharp": "^1.6.16",
```
